tests/distributions/test_omt_mvn.py

```diff
@@ -18,31 +17,81 @@ def analytic_grad(L11=1.0, L22=1.0, L21=1.0, omega1=1.0, omega2=1.0):
 
 
 @pytest.mark.parametrize('L21', [0.4, 1.1])
-@pytest.mark.parametrize('L11', [0.6, 0.95])
-@pytest.mark.parametrize('omega1', [0.5, 0.9])
-@pytest.mark.parametrize('sample_shape', [torch.Size([1000, 1000]), torch.Size([100000])])
-def test_mean_gradient(sample_shape, L21, omega1, L11, L22=0.8, L33=0.9, omega2=0.75):
+@pytest.mark.parametrize('L11', [0.6])
+@pytest.mark.parametrize('omega1', [0.5])
+@pytest.mark.parametrize('sample_shape', [torch.Size([1000, 2000]), torch.Size([200000])])
+@pytest.mark.parametrize('k', [1])
+@pytest.mark.parametrize('mvn_dist', ['OMTMultivariateNormal', 'AVFMultivariateNormal'])
+def test_mean_gradient(mvn_dist, k, sample_shape, L21, omega1, L11, L22=0.8, L33=0.9, omega2=0.75):
+    if mvn_dist == 'OMTMultivariateNormal' and k > 1:
+        return
+
     omega = torch.tensor([omega1, omega2, 0.0])
     loc = torch.zeros(3, requires_grad=True)
     zero_vec = [0.0, 0.0, 0.0]
     off_diag = torch.tensor([zero_vec, [L21, 0.0, 0.0], zero_vec], requires_grad=True)
     L = torch.diag(torch.tensor([L11, L22, L33])) + off_diag
 
-    dist = OMTMultivariateNormal(loc, L)
+    if mvn_dist == 'OMTMultivariateNormal':
+        dist = OMTMultivariateNormal(loc, L)
+    elif mvn_dist == 'AVFMultivariateNormal':
+        CV = torch.tensor(1.1 * torch.rand(2, k, 3), requires_grad=True)
+        dist = AVFMultivariateNormal(loc, L, CV)
+
     z = dist.rsample(sample_shape)
     torch.cos((omega*z).sum(-1)).mean().backward()
 
-    computed_grad = off_diag.grad.data.cpu().numpy()[1, 0]
+    computed_grad = off_diag.grad.clone().data.numpy()[1, 0]
```

Comment:
The `.cpu()` is needed for cuda tests. What is your reason for removing it?

Reply:
i need the clone so that it doesn't get zeroed out

Reply:
I think you could instead `off_diag.grad.cpu()[1, 0].item()`, and that would work in cuda as well